Adding MLAgility models

danielholanda opened the pull request.

ramkrishna2910 commented:
Do we want to add GNNs here? Or wait for the fix to the env?

jeremyfowers commented:
Good opportunity to revisit the directory structure/names?

-`huggingface` is really the pytorch side of huggingface transformers library (rename?)
-`huggingface_tf` is really the Keras side of huggingface transformers library (rename?)
-`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?)
-`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)

danielholanda commented on src/mlagility/parser.py:
```diff
@@ -0,0 +1,55 @@
+
+def parse(valid_args: List):
+
+    # List valid ml-agility args
```
Changed!

danielholanda commented on src/mlagility/parser.py:
```diff
@@ -0,0 +1,55 @@
+
+def parse(valid_args: List):
+
+    # List valid ml-agility args
```
Added!

danielholanda commented:
> Good opportunity to revisit the directory structure/names?
> 
> -`huggingface` is really the pytorch side of huggingface transformers library (rename?) -`huggingface_tf` is really the Keras side of huggingface transformers library (rename?) -`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?) -`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)
> 
> edit: I also want to bring in the "popular on huggingface" models - where do those fit into this dir structure?

@jeremyfowers I addressed all of the suggestions regarding renaming/moving files. 

What do you think we should do about adding the popular models? A more updated list that what we have would be nice. How many models should we add here? Also, is that within the scope of this MR?

danielholanda commented:
> Do we want to add GNNs here? Or wait for the fix to the env?

@ramkrishna2910 I'm ok leaving those here now while we fix the env.

jeremyfowers commented:
> > Do we want to add GNNs here? Or wait for the fix to the env?
> 
> @ramkrishna2910 I'm ok leaving those here now while we fix the env.

Put the GNNs in `gnn/skip`?

jeremyfowers commented:
> > Good opportunity to revisit the directory structure/names?
> > -`huggingface` is really the pytorch side of huggingface transformers library (rename?) -`huggingface_tf` is really the Keras side of huggingface transformers library (rename?) -`torch_hub_vision` is vision models from Torch Hub, but we also have non-vision models from Torch Hub in `torch_hub` (converge?) -`stable_diffusion` is some stable diffusion sub-models from huggingface diffusers library (rename to `diffusers`? we should probably get more models from `diffusers` as well!)
> > edit: I also want to bring in the "popular on huggingface" models - where do those fit into this dir structure?
> 
> @jeremyfowers I addressed all of the suggestions regarding renaming/moving files.
> 
> What do you think we should do about adding the popular models? A more updated list that what we have would be nice. How many models should we add here? Also, is that within the scope of this MR?

Not in the scope of this MR, filed #59. Let's discuss offline :)

jeremyfowers commented on models/graph_convolutions/appnp.py:
```diff
@@ -0,0 +1,35 @@
+# labels: test_group::mlagility name::appnp author::graph_convolutions
```
Move all of these to `graph_convolutions/skip/*`

danielholanda replied:
moved!

jeremyfowers commented on src/mlagility/parser.py:
```diff
@@ -0,0 +1,59 @@
+import argparse
```
add a requirements.txt to `models/.`

danielholanda replied:
Added!
